api/src/main/java/org/xnio/Xnio.java

```diff
@@ -235,6 +235,9 @@ private static Xnio doGetInstance(final String provider, final ServiceLoader<Xni
                 msg.debugf(t, "Skipping non-loadable provider");
             }
         }
+        if (XnioProviderHolder.getProvider() != null) {
```

Comment:
can we have that `provider = XnioProiderHolder.getProvider(); if (provider != null return provider.getInstance();`?